### DISCUSSION TITLE
Error while installing OpenOutpaint extensions on A1111 on Colab

### DISCUSSION THREAD

**Discussion by i-huzaifa-arshad** [TheLastBen/fast-stable-diffusion#1161] [General]
So, I just saw a [post](https://www.reddit.com/r/StableDiffusion/comments/zv8uz5/openoutpaint_a_better_way_to_do_inpainting/?utm_source=share&utm_medium=web2x&context=3) on Reddit about A1111 that it now have an extension "OpenOutpaint". I tried installing it in your A1111 Colab, but getting this error. Can you please go through this issue? So I can enjoy this. Thanks.

Error running install.py for extension /content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/openOutpaint-webUI-extension.
Command: "/usr/bin/python3" "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/openOutpaint-webUI-extension/install.py"
Error code: 1
stdout: <empty>
stderr: Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/extensions/openOutpaint-webUI-extension/install.py", line 3, in <module>
    from launch import run
ModuleNotFoundError: No module named 'launch'

**Comment by TheLastBen** [comment]
I'll check it out, even after the fix you will probably get the "launch" error, just ignore it.

**Reply by i-huzaifa-arshad** [reply]
Alright. Looking forward to your kind response sir.

**Comment by seijihariki** [comment]
Hi, @TheLastBen! I guess you have already given this issue a look, but I will give here a compiled version of the results of an [investigation on our side](https://github.com/zero01101/openOutpaint-webUI-extension/issues/11) regarding installing some extensions to the colab's notebook! (That thread is quite long after all)

Firstly, this seems to mostly be an issue with extensions that require an `install.py` script, such as `dreambooth`, `dreamartist` and of course our `openOutpaint-webUI` extensions, which fail to include the base modules `launch.py` and `modules.py` from `stable-diffusion-webui`'s root. Other extensions that do not need to import local webui files do not seem to be affected, mostly.

From more searching, we found out the errors were due to the assumption from the webui, when running extensions, that the current working directory (`cwd`) was being set to the webui's root. The cwd, when running the `webui.py` command, was, though, the main stablediffusion directory, not the webui's. Noticing that, we added a simple `%cd /content/gdrive/MyDrive/sd/stable-diffusion-webui` to the end of the "Start stable-diffusion" cell, just before running `python3 webui`, and that seems to have fixed all extension compatibility issues we were experiencing. (Including `dreambooth` and  `dreamartist`)

![image](https://user-images.githubusercontent.com/7552751/209936312-2237fb44-0633-4e4b-8d0a-c6f7574c656d.png)

I do not know if this is the optimal solution for this problem, or if any more colab specificities should be considered, but it seems running the webui should be done while on the webui's root as a working directory. If there is any more information you would like, I will promptly provide them!

And great notebook, by the way! Already recommended it to some of my friends!

**Reply by seijihariki** [reply]
@zaif24, for now, adding this `%cd` line should be an okay temporary fix for this problem. Remember to enable the API flag as well if using openOutpaint!

![image](https://user-images.githubusercontent.com/7552751/209936885-379d83fe-2137-49d2-811e-4868323c23da.png)

**Reply by SorenTruelsen** [reply]
Thanks Seijihariki for looking into this. I have not been able to get several of the extensions to work (colab) so I look forward trying this.
